src/VisualStudio/Core/Def/TodoComments/VisualStudioTodoCommentsService.cs

```diff
@@ -77,15 +77,20 @@ public void Dispose()
         void IEventListener<object>.StartListening(Workspace workspace, object _)
         {
             if (workspace is VisualStudioWorkspace)
-                _ = StartAsync();
+                _ = StartAsync(workspace);
         }
 
-        private async Task StartAsync()
+        private async Task StartAsync(Workspace workspace)
         {
             // Have to catch all exceptions coming through here as this is called from a
             // fire-and-forget method and we want to make sure nothing leaks out.
             try
             {
+                // Don't bother doing anything until the workspae has actually loaded.  We don't want to add to any
+                // startup costs by doing work too early.
```

Comment:
should we consider not doing anything at all until someone opens the todo comment window?   I don't think I've ever opened it, so for people like me it seems like there is no point in this even running.

Reply:
Yes.  taht woudl make a ton of sense.  let me see what we can do about that.